modules/tinymce/src/core/test/ts/webdriver/PageUpDownKeysTest.ts

```diff
@@ -0,0 +1,88 @@
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('webdriver.tinymce.core.keyboard.PageUpDownKeyTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    add_unload_trigger: false,
+    base_url: '/project/tinymce/js/tinymce',
+    indent: false
+  }, [ Theme ], true);
+  const platform = PlatformDetection.detect();
+  const supportsPageUpDown = !(platform.os.isOSX() || platform.os.isWindows() && platform.browser.isFirefox());
+
+  // It's necessary to skip tests for Mac and Windows Firefox as they don't move the selection when pressing page up/down
+  before(function () {
+    if (!supportsPageUpDown) {
+      this.skip();
+    }
+  });
+
+  context('Page Up', () => {
+    it('TINY-4612: caret should be placed at the start of the line', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p><a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 1 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 0);
+    });
+
+    it('TINY-4612: caret should be placed out of the line element', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p><a href="google.com">link</a></p>');
+      TinySelections.setCursor(editor, [ 0, 0, 0 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 0);
+    });
+
+    it('TINY-4612: "inline_boundaries: false" macOS doesnt move, whereas other OSs do', async () => {
+      const editor = hook.editor();
+      editor.settings.height = 400;
+      editor.settings.inline_boundaries = false;
+      editor.setContent('<p>test<a href="google.com">link</a>text</p>');
+      TinySelections.setCursor(editor, [ 0, 2 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageUp') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0, 0 ], 0);
+      delete editor.settings.inline_boundaries;
+    });
+  });
+
+  context('Page Down', () => {
+    it('TINY-4612: caret should be placed at the end of the line', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p>text<a href="google.com">link</a></p>');
+      TinySelections.setCursor(editor, [ 0, 0 ], 0);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 2);
+    });
+
+    it('TINY-4612: caret should be placed out of the line element', async () => {
+      const editor = hook.editor();
+      editor.setContent('<p><a href="google.com">link</a></p>');
+      TinySelections.setCursor(editor, [ 0, 0, 0 ], 2);
+      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);
+
+      TinyAssertions.assertCursor(editor, [ 0 ], 1);
+    });
+
+    it('TINY-4612: "inline_boundaries: false" macOS doesnt move, whereas other OSs do', async () => {
```

Comment:
This test doesn't test what's needed and the title doesn't match what's being tested sorry. What we need to test is that the selection doesn't get moved outside the anchor by our custom code when inline boundaries are disabled.

As such, it probably needs to be something like this:
```
    it('TINY-4612: "inline_boundaries: false" cursor does not move outside anchor', async () => {
      const editor = hook.editor();
      editor.settings.height = 400;
      editor.settings.inline_boundaries = false;
      editor.setContent('<p>test<a href="google.com">link</a></p>');
      TinySelections.setCursor(editor, [ 0, 0 ], 0);
      await RealKeys.pSendKeysOn('iframe => body', [ RealKeys.combo({}, 'PageDown') ]);

      TinyAssertions.assertCursor(editor, [ 0, 1, 0 ], 4);
      delete editor.settings.inline_boundaries;
    });
```

You'll see in the above that we make sure a link is at the end and that the placement is where the browser moved it to (eg inside the anchor).

This issue also exists in the equivalent `Page Up` test, so please be sure to update it to do a similar thing as well.